src/libexpr/value-to-xml.cc

```diff
@@ -170,6 +170,11 @@ static void printValueAsXML(
 
     case nThunk:
         doc.writeEmptyElement("unevaluated");
+        break;
+
+    case nFailed:
+        doc.writeEmptyElement("failed");
```

Comment:
Let's not break this code. The XML stuff is rather legacy anyway.

```suggestion
        // Historically, a tried and then ignored value (e.g. through tryEval) was
        // reverted to the original thunk.
        doc.writeEmptyElement("unevaluated");
```